api/vcenter/types.go

```diff
@@ -0,0 +1,56 @@
+// Package vcenter contains vCenter object types.
+package vcenter
+
+// Account contains vCenter account details.
+type Account struct {
+	// Insecure controls whether to validate the vCenter server's certificate.
+	Insecure bool `json:"insecure" yaml:"insecure"`
+
+	// Password is the vCenter password.
+	Password string `json:"password" yaml:"password"`
+
+	// Username is the vCenter username.
+	Username string `json:"username" yaml:"username"`
+
+	// Host is the vCenter URL.
+	Host string `json:"host" yaml:"host"`
+}
+
+// Entity represents a vCenter entity, referenceable via govmomi.
+type Entity int
+
+// nolint:revive
+const (
+	Cluster Entity = iota
+	Datacenter
+	Datastore
+	Folder
+	Host
+	Network
+	ResourcePool
+	VApp
+	VCenterRoot
+	VDS
+	VM
+)
+
+// String converts an Entity to a string.
+func (e Entity) String() string {
+	names := []string{
+		"cluster",
+		"datacenter",
+		"datastore",
+		"folder",
+		"host",
+		"network",
+		"resourcepool",
+		"vapp",
+		"",
```

Comment:
Just double checking with you that the vcenter root string value should be "" as opposed to something like "root"

Reply:
I'm actually not sure yet, but it isn't implemented either. I'll fix it later today when I actually add support for that entity type.